File: gbsuftblai-back/src/main/kotlin/com/kata/gbsuftblai/services/GbsuFtbLaiService.kt
package com.kata.gbsuftblai.services

import org.springframework.stereotype.Component

@Component
class GbsuFtbLaiService {

    fun convertNumber(inputNumber: String): String {
    	val inputAsInt = inputNumber.toInt()
    	var result = ""
    	if(inputAsInt%3 == 0) {
     	   result += "Gbsu"
    	}
    	if(inputAsInt%5 == 0) {
        	result += "Ftb"
    	}
    	for (i in 0..inputNumber.length-1) {
        	if(inputNumber[i] == '3') {
            	result += "Gbsu"
        	} else if(inputNumber[i] == '5') {
            	result += "Ftb"
        	} else if(inputNumber[i] == '7') {
            	result += "Lai"
        	}
    	}
    	if(result == "") {
        	result = inputNumber
    	}
    	return result
    }

}

File: gbsuftblai-ui/src/app/gbsu-ftb-lai.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Result } from './model/result'

@Injectable({
  providedIn: 'root'
})
export class GbsuFtbLaiService {

  constructor(private http:HttpClient) { }

  convertNumber(input: string): Observable<Result> {
    return this.http.get<Result>('http://localhost:8080/gbsu-ftb-lai/' + input);	
  }

}
